modules/c1xBidAdapter.js

```diff
@@ -0,0 +1,207 @@
+// const PIXEL_ENDPOINT = '//px.c1exchange.com/pubpixel/';
+const LOG_MSG = {
+  invalidBid: 'C1X: [ERROR] bidder returns an invalid bid',
+  noSite: 'C1X: [ERROR] no site id supplied',
+  noBid: 'C1X: [INFO] creating a NO bid for Adunit: ',
+  bidWin: 'C1X: [INFO] creating a bid for Adunit: '
+};
+
+/**
+ * Adapter for requesting bids from C1X header tag server.
+ * v3.1 (c) C1X Inc., 2018
+ */
+
+export const c1xAdapter = {
+  code: BIDDER_CODE,
+  supportedMediaTypes: [BANNER],
+
+  /**
+   * Determines whether or not the given bid request is valid.
+   *
+   * @param {object} bid The bid to validate.
+   * @return boolean True if this is a valid bid, and false otherwise.
+  */
+  // check the bids sent to c1x bidder
+  isBidRequestValid: function (bid) {
+    if (bid.bidder !== BIDDER_CODE || typeof bid.params === 'undefined') {
+      return false;
+    }
+    if (typeof bid.params.placementId === 'undefined') {
+      return false;
+    }
+    return true;
+  },
+
+  /**
+    * Make a server request from the list of BidRequests.
+    *
+    * @param {BidRequest[]} validBidRequests A non-empty list of valid bid requests that should be sent to the Server.
+    * @return ServerRequest Info describing the request to the server.
+   */
+  buildRequests: function(validBidRequests, bidderRequest) {
+    let payload = {};
+    let tagObj = {};
+    // let pixelUrl = '';
+    let bidRequest = [];
+    const adunits = validBidRequests.length;
+    const rnd = new Date().getTime();
+    const c1xTags = validBidRequests.map(bidToTag);
+    const bidIdTags = validBidRequests.map(bidToShortTag); // include only adUnitCode and bidId from request obj
+
+    // flattened tags in a tag object
+    tagObj = c1xTags.reduce((current, next) => Object.assign(current, next));
+    // const _bidderSettings = bidderSettings.getSettings();
+    // const pixelId = _bidderSettings.c1x.pixelId || '';
+    // const useSSL = document.location.protocol;
+    // const endpoint = _bidderSettings.c1x.endpoint;
+
+    payload = {
+      adunits: adunits.toString(),
+      rnd: rnd.toString(),
+      response: 'json',
+      compress: 'gzip'
+    }
+
+    // for GDPR support
+    if (bidderRequest && bidderRequest.gdprConsent) {
+      payload['consent_string'] = bidderRequest.gdprConsent.consentString;
+      payload['consent_required'] = (typeof bidderRequest.gdprConsent.gdprApplies === 'boolean') ? bidderRequest.gdprConsent.gdprApplies.toString() : 'true'
+      ;
+    }
+
+    // if (pixelId) {
+    //   pixelUrl = (useSSL ? 'https:' : 'http:') + PIXEL_ENDPOINT + pixelId;
+    //   if (payload.consent_required) {
+    //     pixelUrl += '&gdpr=' + (bidderRequest.gdprConsent.gdprApplies ? 1 : 0);
+    //     pixelUrl += '&consent=' + encodeURIComponent(bidderRequest.gdprConsent.consentString || '');
+    //   }
+    //   userSync.registerSync('image', BIDDER_CODE, pixelUrl);
+    // }
+
+    // tagObj['site'] = _bidderSettings.c1x.siteId | '';
```

Comment:
Is this necessary to leave in your adapter?

Reply:
Removed the comment lines from the file cc @vishnu1211